cmd/fyne/internal/commands/mock_runner_test.go

```diff
@@ -0,0 +1,101 @@
+package commands
+
+import (
+	"os"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockExpectedValue struct {
+	dir   interface{}
+	env   []string
+	osEnv bool
+	args  []string
+}
+
+type mockReturnedValue struct {
+	ret []byte
+	err error
+}
+
+type mockExpectedCall struct {
+	dirSet bool
+	envSet bool
+}
+
+type mockRunOutputValue struct {
+	mockExpectedValue
+	mockReturnedValue
+	mockExpectedCall
+}
+
+type testCommandCall struct {
+	calls []mockRunOutputValue
+	index int
+	t     *testing.T
+}
+
+func (t *testCommandCall) runOutput(args ...string) ([]byte, error) {
+	// Check that we have less than the expected number of call
```

Comment:
There are a lot of checks for t.calls size and t.index being in-sync.
It got me wondering... The mock has a different ordinance value to the real version (i.e. n calls instead of 1)
would it not be simpler if the `testCommandCall` was, as the name implies, singular?

Reply:
I think it will be impossible/limited to use if it was only testing one call as that means we would for example not be able to test build() which might call go once or twice.

Reply:
I think I understand it. 
Should we not be checking then that t.index is exactly 1 more than the length of t.calls?

Reply:
runOutput is called for each command that is to be executed. t.index is the number of call that did happen so far, starting from 0. t.calls is the array that contain all the expected call. So t.index must be within t.call for runOutput to be a valid call which mean t.index < len(t.calls).